### ISSUE TITLE
Signup: Blank "New Account" screen and "Change Username" errors

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-Android#7321]
### Expected behavior

I expect the "New Account" screen to always get pre-populated with default values, or clearly explain if there's an error in the signup process.

### Actual behavior

After using the magic signup link, I saw a popup with a progress spinner on the New Account screen (I think it said "Updating account").

When I tapped on the screen that popup went away but the fields on the New Account screen were blank. Tapping on the Username field brought up this error:

> An error occurred while retrieving username suggestions.

The New Account screen never changed, but I was able to tap "Continue" to move on and I saw on the Me tab that I'd been the default username and my display name matched my username.

Here's the activity log from that process:

```
38 - [Feb-28 10:19 NUX] Google API client connected
39 - [Feb-28 10:19 UTILS] App comes from background
40 - [Feb-28 10:19 STATS] Number of old stats deleted : 0
41 - [Feb-28 10:19 NUX] SignupEmailFragment: Google API client connected
42 - [Feb-28 10:19 NUX] SignupEmailFragment: Google API client connected
43 - [Feb-28 10:19 UTILS] App goes to background
44 - [Feb-28 10:19 MAIN] ConnectionChangeReceiver successfully unregistered
45 - [Feb-28 10:19 UTILS] App comes from background
46 - [Feb-28 10:19 API] Dispatching action: AccountAction-IS_AVAILABLE_EMAIL
47 - [Feb-28 10:19 API] Dispatching action: AccountAction-CHECKED_IS_AVAILABLE
48 - [Feb-28 10:19 API] Dispatching action: AuthenticationAction-SEND_AUTH_EMAIL
49 - [Feb-28 10:19 API] Dispatching action: AuthenticationAction-SENT_AUTH_EMAIL
50 - [Feb-28 10:20 API] Dispatching action: AccountAction-IS_AVAILABLE_EMAIL
51 - [Feb-28 10:20 API] Dispatching action: AccountAction-CHECKED_IS_AVAILABLE
52 - [Feb-28 10:20 API] Dispatching action: AuthenticationAction-SEND_AUTH_EMAIL
53 - [Feb-28 10:20 API] Dispatching action: AuthenticationAction-SENT_AUTH_EMAIL
54 - [Feb-28 10:20 UTILS] App goes to background
55 - [Feb-28 10:20 MAIN] ConnectionChangeReceiver successfully unregistered
56 - [Feb-28 10:20 NUX] SignupEmailFragment: Google API client connected
57 - [Feb-28 10:20 API] Dispatching action: AccountAction-UPDATE_ACCESS_TOKEN
58 - [Feb-28 10:20 UTILS] App comes from background
59 - [Feb-28 10:20 UTILS] trackLastActivity, activityId: My Site
60 - [Feb-28 10:20 API] Dispatching action: AccountAction-FETCH_ACCOUNT
61 - [Feb-28 10:20 NOTIFS] notifications update service > created
62 - [Feb-28 10:20 READER] reader post list > newInstance (tag)
63 - [Feb-28 10:20 READER] reader post list > creating post adapter
64 - [Feb-28 10:20 NOTIFS] notifications update service > completed
65 - [Feb-28 10:20 READER] reader post list > updating tag followed-sites, updateAction=REQUEST_NEWER
66 - [Feb-28 10:20 API] Dispatching action: AccountAction-FETCHED_ACCOUNT
67 - [Feb-28 10:20 NOTIFS] notifications update service > destroyed
68 - [Feb-28 10:20 READER] reader post service > created
69 - [Feb-28 10:20 READER] reader post service > destroyed
70 - [Feb-28 10:20 READER] reader post list > saving instance state
71 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCH_USERNAME_SUGGESTIONS
72 - [Feb-28 10:21 API] Volley error on https://public-api.wordpress.com/wpcom/v2/users/username/suggestions/?name=&locale=en_US - exception: null
73 - [Feb-28 10:21 API] StackTrace: com.android.volley.ServerError
at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:179)
at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)

74 - [Feb-28 10:21 API] {"code":"rest_no_name","message":"A name from which to derive username suggestions is required.","data":{"status":400}}
75 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCHED_USERNAME_SUGGESTIONS
76 - [Feb-28 10:21 API] onUsernameSuggestionsFetched: REST_NO_NAME - A name from which to derive username suggestions is required.
77 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCH_USERNAME_SUGGESTIONS
78 - [Feb-28 10:21 API] Volley error on https://public-api.wordpress.com/wpcom/v2/users/username/suggestions/?name=&locale=en_US - exception: null
79 - [Feb-28 10:21 API] StackTrace: com.android.volley.ServerError
at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:179)
at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)

80 - [Feb-28 10:21 API] {"code":"rest_no_name","message":"A name from which to derive username suggestions is required.","data":{"status":400}}
81 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCHED_USERNAME_SUGGESTIONS
82 - [Feb-28 10:21 API] onUsernameSuggestionsFetched: REST_NO_NAME - A name from which to derive username suggestions is required.
83 - [Feb-28 10:21 UTILS] trackLastActivity, activityId: My Site
84 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCH_ACCOUNT
85 - [Feb-28 10:21 READER] reader post list > resumed from paused state
86 - [Feb-28 10:21 READER] reader post list > current tag no longer valid
87 - [Feb-28 10:21 API] Dispatching action: AccountAction-FETCHED_ACCOUNT
88 - [Feb-28 10:21 UTILS] trackLastActivity, activityId: Me
89 - [Feb-28 10:21 UTILS] trackLastActivity, activityId: Help Screen
90 - [Feb-28 10:21 READER] reader post list > saving instance state
91 - [Feb-28 10:21 UTILS] App goes to background
```

### Screenshots

![2018-02-28 10 21 22](https://user-images.githubusercontent.com/8658164/36782907-55444314-1c72-11e8-9055-1091274af568.png)
![2018-02-28 10 21 16](https://user-images.githubusercontent.com/8658164/36782909-563fce96-1c72-11e8-9c21-92375fe3cdc4.png)

### Steps to reproduce the behavior

This occurred when I signed up with email, but I haven't been able to reproduce the behavior. However, I wanted to raise the issue in case others run into it or there's a way to handle it even without reproducible steps — it's a very confusing issue for a new user to face.

##### Tested on Nexus 9, Android 7.1.1

**Comment by elibud** [comment]
@theck13 please look into this one

**Comment by theck13** [comment]
The ***Change Username*** screen shows an error because the query is seeded by the values on the ***New Account*** screen.  Since there are none, the query is empty causing the failure.

I'm not able to reproduce this.  I've tried canceling the progress dialog using the magic link from an email, but the values are eventually populated even with a very slow network connection.  Nothing in the application log looks odd.  Are there any other testing details that might help?

**Comment by rachelmcr** [comment]
I've done more testing and I think I've found it: It occurs when I'm already logged in to a self-hosted site and then sign up for WordPress.com. These steps are reliably reproducing the issue for me:

1. Start completely logged out of the app.
2. Log in to a self-hosted (non-Jetpack) site in the app.
3. Go to the Me tab and select "Log in to WordPress.com."
4. Go through the signup flow (I used email signup).

When I follow these steps, after using the magic link I'm taken to the "New account" screen with an "Updating account..." dialog. When I tap on the screen the dialog is dismissed but the fields are blank.
